### ISSUE TITLE
Empty attribute

### ISSUE THREAD

**Issue by kuon** [mdgriffith/stylish-elephants#38]
To write this helper:

```
whenAttr : Bool -> Attribute -> Attribute
whenAttr cond attr =
  if cond then
    attr
  else
    attribute "data-empty" ""
```

I need to use a hack `data-empty`, I wonder if it would be possible to add a `empty` attribute that will be removed by SE.

**Comment by kuon** [comment]
For use case, I use it for things like setting conditional opacity on some element if they are enabled or not. I also use it for conditional event handlers.

**Comment by tryge** [comment]
Realworld Use Case (edited to remove some attributes to avoid an unreadably long line...):

```
(height (px 16) :: (if focussed || value /= "" then [] else [moveDown 20])) ++ defaultLabelAttrs
```

**Comment by mdgriffith** [comment]
So, for the usecases mentioned, they can all be nullified with a value.

```elm
moveDown <|
    if focussed || value /= "" then
        0
    else
        20

alpha <|
    if condition then
        0
    else
        1

onClick <|
    if enabled then
          MyMsg
    else
          NoOp
```

This gets away from having to use something like `List.concat`.

**Comment by kuon** [comment]
True, except for the `onClick` it will fire an event, depending on the use case it can be a performance issue, like tracking scroll only when some element is visible.

**Comment by kuon** [comment]
But if you feel against it, my hack is perfectly fine, I don't think it is a priority.

**Comment by mdgriffith** [comment]
Hmm, we could have an event handler that just failed to decode and so never sent it's message.

Like `Events.never` or something.

**Comment by kuon** [comment]
For `onScroll` just having an event handler attached has a big perf impact. There is also the cases were attaching an event handler can heavily impact the browser.

**Comment by tryge** [comment]
@mdgriffith it's true that these use cases can be nullified with a value, and I actually refactored today a little and it is definitely an improvement, so thank you :-) 

Nevertheless I like the `Attributes.none` approach more because in my opinion it's more readable and easier to recognise the pattern, whereas the values that nullify attributes differ depending on the use case.

**Comment by norpan** [comment]
My suggestion is to also allow attribute lists within the attribute list. This is useful for instance if you have lists of styles you want to apply. Of course, it's always possible to use concatenation instead.

**Comment by kuon** [comment]
I just realized that inherited attributes are a problem. For example, you cannot do:

`[if selected then Font.bold else Attribute.empty]`

I cannot replace `Attribute.empty` because I expect the font weight to be inherited. It can be font size, font weight, text color. Maybe some other. Some properties make the most sense when inherited.
